src/sql-parser/src/parser.rs

```diff
@@ -4086,10 +4119,76 @@ impl<'a> Parser<'a> {
                 }))
             }
             ObjectType::Cluster => self.parse_alter_cluster(object_type),
+            ObjectType::ReplicaSet => {
+                let if_exists = self.parse_if_exists().map_no_statement_parser_err()?;
+                let name = self
+                    .parse_cluster_item_name()
+                    .map_no_statement_parser_err()?;
+                let action = self
+                    .expect_one_of_keywords(&[OWNER, RENAME, SET, RESET])
+                    .map_no_statement_parser_err()?;
+                match action {
+                    OWNER => {
+                        self.expect_keyword(TO).map_no_statement_parser_err()?;
+                        let new_owner = self
+                            .parse_identifier()
+                            .map_parser_err(StatementKind::AlterOwner)?;
+                        let name = UnresolvedObjectName::ClusterItem(name);
+                        Ok(Statement::AlterOwner(AlterOwnerStatement {
+                            object_type,
+                            if_exists,
+                            name,
+                            new_owner,
+                        }))
+                    }
+                    RENAME => {
+                        self.expect_keyword(TO).map_no_statement_parser_err()?;
+                        let to_item_name = self
+                            .parse_identifier()
+                            .map_parser_err(StatementKind::AlterObjectRename)?;
+                        let name = UnresolvedObjectName::ClusterItem(name);
+                        Ok(Statement::AlterObjectRename(AlterObjectRenameStatement {
+                            object_type,
+                            if_exists,
+                            name,
+                            to_item_name,
+                        }))
+                    }
+                    RESET => {
+                        self.expect_token(&Token::LParen)
+                            .map_parser_err(StatementKind::AlterReplicaSet)?;
+                        let names = self
+                            .parse_comma_separated(Parser::parse_cluster_option_name)
+                            .map_parser_err(StatementKind::AlterReplicaSet)?;
+                        self.expect_token(&Token::RParen)
+                            .map_parser_err(StatementKind::AlterReplicaSet)?;
+                        Ok(Statement::AlterReplicaSet(AlterReplicaSetStatement {
+                            if_exists,
+                            name,
+                            action: AlterReplicaSetAction::ResetOptions(names),
+                        }))
+                    }
```

Comment:
Covered by unit tests but not end-to-end tests according to coverage data. Can we add an end-to-end test that covers this code?